builders/Dockerfile.centos7

```diff
@@ -8,7 +9,7 @@ ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH
 
-RUN wget "https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init" \
+RUN wget "https://static.rust-lang.org/rustup/dist/${arch}-unknown-linux-gnu/rustup-init" \
```

Comment:
I don't understand how this can work. Cent OS 7 only supports x86_64: https://en.wikipedia.org/wiki/CentOS#CentOS_version_7. How can this run an aarch64 binary? This is an ARM binary in an x86 OS. But then again there is ARM64 here: https://hub.docker.com/layers/centos/library/centos/centos7/images/sha256-73f11afcbb50d8bc70eab9f0850b3fa30e61a419bc48cf426e63527d14a8373b?context=explore.

I probably don't understand docker.

Reply:
I don't think it only supports x86_64. There are mirrors for Centos7 for aarch64, and your wiki mentions the following:

**As of December 2015, AltArch releases of CentOS 7 are available for the ARMv7hl and [AArch64](https://en.wikipedia.org/wiki/AArch64) variants of the [ARM architecture](https://en.wikipedia.org/wiki/ARM_architecture)**

I'm using an M1 so I am running on an ARM64 machine. I believe docker chooses the appropriate image. In my case this would be an aarch64 version of centos.

I encountered the opposite problem which it was trying a run an x86_64 binary on an aarch64.

Reply:
> As of December 2015, AltArch releases of CentOS 7 are available for the ARMv7hl and [AArch64](https://en.wikipedia.org/wiki/AArch64) variants of the [ARM architecture](https://en.wikipedia.org/wiki/ARM_architecture)

Nice. This was the missing piece. This also affects the discussion in https://github.com/CosmWasm/wasmvm/issues/293.

Reply:
Also easy to verify:
```
❯ docker run --rm -it --platform linux/arm64 centos:7 uname -m
aarch64

❯ docker run --rm -it --platform linux/amd64 centos:7 uname -m
x86_64
```

There's a multi-arch image published for `centos:7`:
```
❯ docker manifest inspect centos:7
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 529,
         "digest": "sha256:dead07b4d8ed7e29e98de0f4504d87e8880d4347859d839686a31da35a3b532f",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      /.../
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 530,
         "digest": "sha256:73f11afcbb50d8bc70eab9f0850b3fa30e61a419bc48cf426e63527d14a8373b",
         "platform": {
            "architecture": "arm64",
            "os": "linux",
            "variant": "v8"
         }
      }
   ]
}
```